### PR TITLE
Fix EnC issue with out of sync DocumentId

### PR DESCRIPTION
**Pull request by drognanar**

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs [hunk 1]
```diff
@@ -590,10 +590,13 @@ private void AddActiveStatements(Solution solution, ShellInterop.ENC_ACTIVE_STAT
 
                 var flags = (ActiveStatementFlags)vsActiveStatement.ASINFO;
 
-                IVisualStudioHostDocument vsDocument = _vsProject.GetCurrentDocumentFromPath(vsActiveStatement.filename);
-                if (vsDocument != null)
+                // Finds a document id in the solution with the specified file path.
+                DocumentId documentId = solution.GetDocumentIdsWithFilePath(vsActiveStatement.filename)
+                    .Where(dId => dId.ProjectId == _vsProject.Id).FirstOrDefault();
```
Comment (amcasey) [inline src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs]: There should be zero or one, no?  Depending on IDE conventions, consider using `SingleOrDefault`.

**Comment by amcasey** [comment]
test vsi please

**Comment by amcasey** [comment]
:+1:

**Comment by drognanar** [comment]
test prtest/lin/dbg/unit32 please
test prtest/mac/dbg/unit32 please

**Comment by drognanar** [comment]
test prtest/mac/dbg/unit32 please

**Comment by drognanar** [comment]
test prtest/mac/dbg/unit32 please

**Pull request merged**